rnd/autogpt_builder/src/components/Flow.tsx

```diff
@@ -295,13 +280,18 @@ const FlowEditor: React.FC<{
       return {};
     }
 
-    const getNestedData = (schema: ObjectSchema, values: { [key: string]: any }): { [key: string]: any } => {
+    const getNestedData = (
+      schema: BlockIOSubSchema, values: { [key: string]: any }
+    ): { [key: string]: any } => {
       let inputData: { [key: string]: any } = {};
 
-      if (schema.properties) {
+      if ("properties" in schema) {
```

Comment:
In cases like this it's common to use type guards for readability, type safety, and a bit of assistance from Intellisense in the future. 

An alternative if you are up for it is to add something like this to your types.ts:

```typescript
// Type guard for BlockIOObjectSubSchema
export function isBlockIOObjectSubSchema(schema: BlockIOSubSchema): schema is BlockIOObjectSubSchema {
  return schema.type === "object" && "properties" in schema;
}

// Type guard for BlockIOKVSubSchema
export function isBlockIOKVSubSchema(schema: BlockIOSubSchema): schema is BlockIOKVSubSchema {
  return schema.type === "object" && "additionalProperties" in schema;
}
```

Then you can import it and use it like this in your  Flow.tsx

```suggestion
     if (isBlockIOObjectSubSchema(schema)) 
```

and 

```typescript
if (isBlockIOKVSubSchema(schema))
```

Reply:
All together it would look something like:

```typescript
const getNestedData = (
  schema: BlockIOSubSchema, values: { [key: string]: any }
): { [key: string]: any } => {
  let inputData: { [key: string]: any } = {};

  if (isBlockIOObjectSubSchema(schema)) {
    Object.keys(schema.properties).forEach((key) => {
      if (values[key] !== undefined) {
        inputData[key] = isBlockIOObjectSubSchema(schema.properties[key]) || isBlockIOKVSubSchema(schema.properties[key])
          ? getNestedData(schema.properties[key], values[key])
          : values[key];
      }
    });
  }

  if (isBlockIOKVSubSchema(schema)) {
    inputData = { ...inputData, ...values };
  }

  return inputData;
};
```